packages/entity-browser/src/routes/detail/components/DetailForm/DetailForm.spec.js

```diff
@@ -104,6 +126,74 @@ describe('entity-browser', () => {
         expect(wrapper.find('form')).to.have.length(1)
         expect(wrapper.find(Field)).to.have.length(2)
       })
+
+      it('should request user confirmation when touched form is left', () => {
+        const store = createStore(combineReducers({form: formReducer}))
+
+        const block = sinon.spy()
+
+        const fakeContext = {router: {block}}
+        const contextTypes = {router: React.PropTypes.object}
+
+        const formComponent = context.wrapWithContext(fakeContext, contextTypes,
+          <DetailForm
+            submitting={false}
+            submitForm={EMPTY_FUNC}
+            formDefinition={SIMPLE_FORM_DEFINITION}
+            entity={SIMPLE_ENTITY}
+            entityModel={{}}
+            logError={EMPTY_FUNC}
+            loadRelationEntities={EMPTY_FUNC}
+            relationEntities={{}}
+            form="detailForm"
+            intl={IntlStub}
+          />
+        )
+
+        mount(
+          <Provider store={store}>
+            {formComponent}
+          </Provider>
+        )
+
+        // touch a field to enable the prompt -> context.history.block must be called
+        store.dispatch(touch('detailForm'), 'firstname')
+
+        expect(block).to.be.calledWith('client.entity-browser.confirmTouchedFormLeave')
+      })
+
+      it('should not request user confirmation when untouched form is left', () => {
+        const store = createStore(combineReducers({form: formReducer}))
+
+        const block = sinon.spy()
+
+        const fakeContext = {router: {block}}
+        const contextTypes = {router: React.PropTypes.object}
+
+        const formComponent = context.wrapWithContext(fakeContext, contextTypes,
+          <DetailForm
+            submitting={false}
+            submitForm={EMPTY_FUNC}
+            formDefinition={SIMPLE_FORM_DEFINITION}
+            entity={SIMPLE_ENTITY}
+            entityModel={{}}
+            logError={EMPTY_FUNC}
+            loadRelationEntities={EMPTY_FUNC}
+            relationEntities={{}}
+            form="detailForm"
+            intl={IntlStub}
+          />
+        )
+
+        mount(
+          <Provider store={store}>
+            {formComponent}
+          </Provider>
+        )
+
+        // form is untouched -> context.history.block must not be called
+        expect(block).not.to.be.calledWith('client.entity-browser.confirmTouchedFormLeave')
```

Comment:
i think expect(block).not.to.be.called() would be better since we dont care about the argument in this case.